src/components/button/button.tsx

```diff
@@ -47,6 +58,19 @@ const Button: React.FunctionComponent<ButtonProps> = props => {
 		}
 	}
 
+	const renderIcon = () => {
+		let iconNode = null
+		if (loading) {
+			iconNode = <LoadingOutlined />
+		}
+
+		if (icon) {
+			iconNode = icon
+		}
+
+		return iconNode ? <span className="icon">{iconNode}</span> : null
```

Comment:
Check antd for reference 

https://github.com/ant-design/ant-design/blob/master/components/button/button.tsx